### PR TITLE
Tweak: Google Maps Widget - Added support for the new Google Maps Embed API (ED-3394) (#15090)

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Google Maps Widget: Added support for the new Google Maps Embed API with API Key usage.

## Quality assurance

- [X] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: includes/widgets/google-maps.php [hunk 1]
```diff
@@ -256,12 +271,24 @@ protected function render() {
 			$settings['zoom']['size'] = 10;
 		}
 
-		printf(
-			'<div class="elementor-custom-embed"><iframe frameborder="0" scrolling="no" marginheight="0" marginwidth="0" src="https://maps.google.com/maps?q=%1$s&amp;t=m&amp;z=%2$d&amp;output=embed&amp;iwloc=near" title="%3$s" aria-label="%3$s"></iframe></div>',
-			rawurlencode( $settings['address'] ),
-			absint( $settings['zoom']['size'] ),
-			esc_attr( $settings['address'] )
-		);
+		$api_key = esc_html( get_option( 'elementor_google_maps_api_key' ) );
```
Comment (kobizz) [inline includes/widgets/google-maps.php:274]: Do the escape where you print it

File: includes/settings/settings.php [hunk 2]
```diff
@@ -460,7 +460,21 @@ protected function create_tabs() {
...
+				'sections' => [
+					'google_maps' => [
+						'label' => __( 'Google Maps Embed API', 'elementor' ),
+						'fields' => [
+							'google_maps_api_key' => [
+								'label' => __( 'API Key', 'elementor' ),
+								'field_args' => [
+									'class' => 'elementor_google_maps_api_key',
+									'type' => 'text',
+									'desc' => __( 'Enter a valid Google Maps Embed API key. For more details, visit Google Maps Embed API\'s <a href="https://developers.google.com/maps/documentation/embed/get-api-key" target="_blank">Using API Keys</a> page.', 'elementor' ),
```
Comment (kobizz) [inline includes/settings/settings.php:472]: Change these texts

File: includes/widgets/google-maps.php [hunk 3]
```diff
@@ -256,12 +271,24 @@ protected function render() {
...
+
+		if ( $api_key ) {
+			printf(
```
Comment (kobizz) [inline includes/widgets/google-maps.php:277]: The only difference is just the url

**Review by kobizz — CHANGES_REQUESTED** [review]

File: includes/widgets/google-maps.php (hunk 3 above)
Comment (udidol) [reply includes/widgets/google-maps.php]: Done.

File: includes/settings/settings.php (hunk 2 above)
Comment (udidol) [reply includes/settings/settings.php]: Done.

File: includes/widgets/google-maps.php (hunk 1 above)
Comment (udidol) [reply includes/widgets/google-maps.php:278]: Done.

File: includes/widgets/google-maps.php [hunk 4]
```diff
@@ -104,6 +104,21 @@ protected function register_controls() {
 			]
 		);
 
+		if ( Plugin::$instance->editor->is_edit_mode() ) {
+			$api_key = esc_html( get_option( 'elementor_google_maps_api_key' ) );
```
[the remark this answers is not in the document]
Comment (udidol) [reply includes/widgets/google-maps.php]: Done.

**Review by kobizz — APPROVED** [review]

**Pull request merged**